pkg/client/postgres/tasks.go

```diff
@@ -95,24 +136,69 @@ func (db *PostgresDB) ScheduleBackup(schedule string) error {
 	return db.client.Apply(db.Namespace, job)
 }
 
-func (db *PostgresDB) ListBackups() ([]string, error) {
-	var backups []string
-	return backups, nil
+func (db *PostgresDB) ListBackups(s3Bucket string) error {
+	opClusterEnv := *db.opClusterEnv
+
+	var backupS3Bucket string
+	if s3Bucket != "" {
+		backupS3Bucket = s3Bucket
+	} else {
+		backupS3Bucket = db.backupBucket
+	}
+
+	job := kommons.
+		Deployment("list-backups-"+db.Name+"-"+utils.ShortTimestamp(), string(opClusterEnv["BACKUP_IMAGE"])).
+		EnvVarFromSecret("PGPASSWORD", db.Secret, "password").
+		EnvVars(map[string]string{
+			"AWS_ACCESS_KEY_ID":     string(opClusterEnv["AWS_ACCESS_KEY_ID"]),
+			"AWS_SECRET_ACCESS_KEY": string(opClusterEnv["AWS_SECRET_ACCESS_KEY"]),
+			"AWS_DEFAULT_REGION":    string(opClusterEnv["AWS_DEFAULT_REGION"]),
+			"RESTIC_REPOSITORY":     fmt.Sprintf("s3:%s/%s", string(opClusterEnv["AWS_ENDPOINT_URL"]), backupS3Bucket),
+			"RESTIC_PASSWORD":       string(opClusterEnv["BACKUP_PASSWORD"]),
+			"PGHOST":                db.Name,
+		}).
+		Command("/list.sh").
+		AsOneShotJob()
+
+	defer func() {
+		if err := db.client.DeleteByKind("Pod", db.Namespace, job.Name); err != nil {
```

Comment:
Can we not use `ttlSecondsAfterFinished` for the cleanup?

Reply:
The `AsOneShotJob()` (https://github.com/flanksource/kommons/blob/10b3cd835a989749a06aeb564a153881494e88eb/builder.go#L216) returns a pod template while `.spec.ttlSecondsAfterFinished` is available to Job object only. Unless we change the `AsOneShotJob()` to generate Job instead of Pod, we can't use `ttlSecondsAfterFinished`

Reply:
It might make sense to change it from generating a pod to job, and/or renaming the existing method to AsOneShotPod